src/structures/gdt.rs

```diff
@@ -93,36 +93,34 @@ impl GlobalDescriptorTable {
         &self.table[..self.next_free]
     }
 
-    const_fn! {
-        /// Adds the given segment descriptor to the GDT, returning the segment selector.
-        ///
-        /// Panics if the GDT has no free entries left.  Without the `const_fn`
-        /// feature, the panic message will be "index out of bounds".
-        #[inline]
-        pub fn add_entry(&mut self, entry: Descriptor) -> SegmentSelector {
-            let index = match entry {
-                Descriptor::UserSegment(value) => self.push(value),
-                Descriptor::SystemSegment(value_low, value_high) => {
-                    let index = self.push(value_low);
-                    self.push(value_high);
-                    index
-                }
-            };
+    /// Adds the given segment descriptor to the GDT, returning the segment selector.
+    ///
+    /// Panics if the GDT has no free entries left.
+    #[inline]
+    #[cfg_attr(feature = "const_fn", rustversion::attr(all(), const))]
```

Comment:
Why not just `#[cfg_attr(feature = "const_fn", const)]`?

Reply:
Because `#[const]` is not a real attribute. `rustversion` has a special case for `const` where it adds the const keyword to the method declaration instead of adding it as an attribute.

I got tripped up by that as well :)

Reply:
Yes, it turns out `#[cfg_attr(feature = "const_fn", rustversion::attr(all(), const))]` works, but the following don't work:

- `#[cfg_attr(feature = "const_fn", const)]`
- `#[rustversion::attr(feature = "const_fn", const)]`
- `#[cfg_attr(feature = "const_fn", rustversion::const)]`

It took a little trial and error to find one that worked.

Reply:
Ah yes, makes sense. Thanks!